Sources/SwiftParser/Lexer/UnicodeScalarExtensions.swift

```diff
@@ -259,3 +246,38 @@ extension UInt8 {
     return self < 0x80 || (self >= 0xC2 && self < 0xF5)
   }
 }
+
+/// Allows direct comparisons between UInt8 and double quoted literals.
+extension UInt8 {
+    /// Basic equality operators
+    @_transparent
+    static func == (i: Self, s: Unicode.Scalar) -> Bool {
```

Comment:
The doc comment applies only to the `==` function, so `Basic equality operators` doesn’t really make sense.